src/include/simeng/RegisterValue.hh

```diff
@@ -129,6 +140,9 @@ class RegisterValue {
   /** The underlying local member value. Aligned to 8 bytes to prevent
    * potential alignment issue when casting. */
   alignas(8) char value[MAX_LOCAL_BYTES];
+
+  /** Switch for different assert checking */
+  bool relaxedFor32bit_;
```

Comment:
Could you please give additional reasoning for the addition of this new class variable and it's usage? 
Also please see comments [1](https://github.com/UoB-HPC/SimEng/pull/317/files#r1309922102) & [2](https://github.com/UoB-HPC/SimEng/pull/317/files#r1309914613) first.